pkg/core/resources/apis/mesh/traffic_route_validator.go

```diff
@@ -0,0 +1,54 @@
+package mesh
+
+import (
+	"fmt"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	"github.com/Kong/kuma/pkg/core/validators"
+)
+
+func (d *TrafficRouteResource) Validate() error {
+	var err validators.ValidationError
+	err.Add(d.validateSources())
+	err.Add(d.validateDestinations())
+	err.Add(d.validateConf())
+	return err.OrNil()
+}
+
+func (d *TrafficRouteResource) validateSources() validators.ValidationError {
+	return ValidateSelectors(validators.RootedAt("sources"), d.Spec.Sources, ValidateSelectorOpts{})
+}
+
+func (d *TrafficRouteResource) validateDestinations() (err validators.ValidationError) {
+	return ValidateSelectors(validators.RootedAt("destinations"), d.Spec.Destinations, ValidateSelectorOpts{
+		SkipRequireAtLeastOneTag: true,
```

Comment:
What is the use case of this flag? I thought we should always require service tag in destinations

Reply:
`TrafficRoute` has a special restriction on `destinations` and I have a proper message for that.

I think, it might be confusing for a user to see both messages at the same time:
* `must have at least one tag` 
and 
* `must consist of exactly one tag "service"`

Reply:
I think it would be fine, but ok :)